File: frontend/src/types/warehouse.ts
export interface Warehouse {
  id: string;
  tenant_id: string;
  name: string;
  address: string | null;
  city: string | null;
  state: string | null;
  country: string | null;
  postal_code: string | null;
  capacity_total: number | null;
  capacity_unit: string | null;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface WarehouseCreate {
  name: string;
  address?: string | null;
  city?: string | null;
  state?: string | null;
  country?: string | null;
  postal_code?: string | null;
  capacity_total?: number | null;
  capacity_unit?: string | null;
  is_active?: boolean;
}

export interface WarehouseUpdate {
  name?: string | null;
  address?: string | null;
  city?: string | null;
  state?: string | null;
  country?: string | null;
  postal_code?: string | null;
  capacity_total?: number | null;
  capacity_unit?: string | null;
  is_active?: boolean | null;
}

File: backend/src/models/warehouse.py
"""
Warehouse model representing physical storage locations.
"""
from sqlalchemy import Column, String, Text, Boolean, Numeric, UniqueConstraint, Index
from sqlalchemy.dialects.postgresql import UUID

from src.models.base import BaseModel


class Warehouse(BaseModel):
    """Warehouse entity representing physical storage locations."""
    
    __tablename__ = "warehouses"
    
    name = Column(String(255), nullable=False)
    address = Column(Text, nullable=True)
    city = Column(String(100), nullable=True)
    state = Column(String(100), nullable=True)
    country = Column(String(100), nullable=True)
    postal_code = Column(String(20), nullable=True)
    capacity_total = Column(Numeric(15, 2), nullable=True)  # Total capacity
    capacity_unit = Column(String(50), nullable=True)  # e.g., 'cubic_meters', 'square_meters'
    is_active = Column(Boolean, nullable=False, default=True, index=True)
    created_by = Column(UUID(as_uuid=True), nullable=True)
    updated_by = Column(UUID(as_uuid=True), nullable=True)
    
    # Constraints
    __table_args__ = (
        UniqueConstraint('tenant_id', 'name', name='uq_warehouses_tenant_name'),
        Index('idx_warehouses_tenant_name', 'tenant_id', 'name', unique=True),
        Index('idx_warehouses_active', 'is_active'),
    )
    
    def __repr__(self):
        return f"<Warehouse(id={self.id}, name={self.name}, is_active={self.is_active})>"
